### ISSUE TITLE
Controls for headless pi?

### ISSUE THREAD

**Issue by watahboy** [scottlawsonbc/audio-reactive-led-strip#78]
I love this project but I'd love the ability to be able to change the visualizations on a standalone pi install while running headless. Is there any way to make a hardware switch to cycle the animations so I I can run the system standalone?

**Comment by Toms42** [comment]
Yes. You can edit the mode just like this: https://github.com/scottlawsonbc/audio-reactive-led-strip/blob/master/python/visualization.py#L251

You'll have to write the supporting software to read a hardware switch and update the mode though of course.

**Comment by watahboy** [comment]
Thanks for pointing that out. Was no trouble at all editing the visualization.py, I guess I missed that line when I was skimming for that option before. Next chance I get I'm going to try to integrate a hardware controller or at the very least some mqtt control so I can run this without having to launch it from putty or add it to crontab. 

Thanks again, I have already been using your version with the overflow fix at 50fps on my rpi3 raspbian pihole and its been stable for hours.

**Comment by Arthur266** [comment]
Hi can you post the overflow fix. It doesn't work for me I bought now a second sound module but it isn't working. Only 3 times a get 50 to 60 fps than all the time overflow. 
Thanks

**Comment by Toms42** [comment]
Yep! Here it is: https://github.com/scottlawsonbc/audio-reactive-led-strip/pull/73

Glad to hear that the fix is working @watahboy!

**Comment by Arthur266** [comment]
Thx :)

**Comment by Arthur266** [comment]
Do you know maybe how can I control a ws2811 led strip with the esp 8266 module. It's only for ws 2812. 
Thanks ))

**Comment by Toms42** [comment]
No, I don't sorry.

**Comment by watahboy** [comment]
Well I've managed to get the three visualizations to load from a USB RF presenters remote by setting the keybinds to three separate visualizations.py files I made with that one line switched out. I was hoping there would be a cleaner way or a command that could toggle the selection but as I am still learning python hopefully I can figure it out eventually. All my attempts so far to even make the script terminate (like ctrl-c) haven't worked since this is running headless and I can not see the script running in CLI.

On the plus side the script launches much faster without the CLI open, I am starting to consider going above 60fps once I nail down these settings. 

If anyone has insight in a graceful way to switch between the visualizations or dare I say other settings available from the gui on a headless pi I'd love to get a little help.

**Comment by microfx** [comment]
Did you manage this? This would make a great addition to this pack - using MIDI or OSC or a physical button would be so nice!
